### PR TITLE
Update & simplify goverter

### PR DESCRIPTION
**Pull request by jmattheis**
I've found your project while looking for projects using goverter. It's great
to see how it's used (:. I've upgraded it to the newest version and simplified
some stuff. See commit messages for explanations.

The last commit is a showcase how some custom code could be converted, I think
both solutions are valid, so I haven't changed this for every occurrence.

I've also created the feature request for regex `goverter:map` in
https://github.com/jmattheis/goverter/issues/122 because this would simplify a
lot of `goverter:map` calls here.

Goverter can be execute with `go generate ./db`

### CODE REVIEW

**Comment by sirgwain** [comment]
Wow! Thank you so much! I was looking for a code generator to copy my database domain objects into the game domain and `goverter` fit the bill perfectly! I noticed you had upgraded and incorporated a number of changes, but I hadn't gotten around to figuring out how to migrate yet. :)

File: db/generated.go [hunk 1]
```diff
@@ -12,14 +13,28 @@ type GameConverter struct{}
...
+		csFleet.Cargo = c.dbFleetToCsCargo((*source))
+		csFleet.Fuel = (*source).Fuel
+		csFleet.Age = (*source).Age
+		csFleet.Tokens = ShipTokensToGameShipTokens((*source).Tokens)
+		csFleet.Heading = ExtendFleetHeading((*source))
+		csFleet.WarpSpeed = (*source).WarpSpeed
+		csFleet.PreviousPosition = ExtendFleetPreviousPosition((*source))
+		csFleet.OrbitingPlanetNum = (*source).OrbitingPlanetNum
+		csFleet.Starbase = (*source).Starbase
+		csFleet.Spec = FleetSpecToGameFleetSpec((*source).Spec)
```
Comment (sirgwain) [inline db/generated.go:30]: I have a question on this generated code. Won't this create a copy of `source` for each variable assignment? 

I think this pointer dereferencing makes a copy of the struct each time. I'm not sure if the go compiler optimizes that sort of thing away. 

```golang
csFleet.PlanetNum = (*source).PlanetNum
csFleet.BaseName = (*source).BaseName
csFleet.Fuel = (*source).Fuel
csFleet.Age = (*source).Age
```

In the previous iteration, the pointer was only dereferenced once when calling 
```golang
c.dbFleetToCsFleet((*source))
```

Comment (jmattheis) [reply db/generated.go:30]: Dereferencing does not copy `source`, dereferencing only gets the underlying value. The code above is actually what happens in the background when you access fields on a struct pointer in go. E.g.
```
type S struct{ Name string }

func main() {
	var s *S = nil
	_ = s.Name
	_ = *s
}
```
Both expressions `s.Name` and `*s` dereference `s` and they also both throw the same nil panic.

See https://stackoverflow.com/a/38443456/4244993

**Review by sirgwain — APPROVED** [review]

File: db/generated.go (hunk 1 above)
Comment (sirgwain) [reply db/generated.go:30]: Oh nice, I didn’t realize that but it makes sense. Thanks!

**Pull request merged**